Update remote-state-data.mdx (#31458)
hashicorp/terraform: website/docs/language/state/remote-state-data.mdx

You can use the `terraform_remote_state` data source without requiring or configuring a provider. It is always available through a built-in provider with the [source address](/language/providers/requirements#source-addresses) `terraform.io/builtin/terraform`. That provider does not include any other resources or data sources.

> ~> **Important:** We recommend using the [`tfe_outputs` data source](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/data-sources/outputs) in the [Terraform Cloud/Enterprise Provider](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs) to access remote state outputs in Terraform Cloud or Terraform Enterprise. The `tfe_outputs` data source is more secure because it requires the Terraform Cloud/Enterprise Provider API token to have "Read outputs only" permission to fetch outputs.

Reviewer: Hi! Thank you for submitting this! I'm thinking through the best way to handle. Specifically, this note is quite long and I'm wondering if we can fix the problem you identified without making it longer. 

I'm wondering if the problematic statement is "it does not require full access". I wonder if instead, we could change this to:

"The `tfe_outputs` data source is more secure because it does not require write access to workspace state to fetch outputs"

This way, we're clarifying that it does require some access, but we're keeping the sentence a bit more concise. Let me know what you think!

Author: Hi @laurapacilio,

I agree that the proposed change makes the sentence more concise and it looks good to me so let's have the documentation updated that way.

Thanks for the help here!

> ~> **Important:** We recommend using the [`tfe_outputs` data source](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/data-sources/outputs) in the [Terraform Cloud/Enterprise Provider](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs) to access remote state outputs in Terraform Cloud or Terraform Enterprise. The `tfe_outputs` data source is more secure because it requires the Terraform Cloud/Enterprise Provider API token to have "Read outputs only" permission to fetch outputs.

Reviewer: ```suggestion
~> **Important:** We recommend using the [`tfe_outputs` data source](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/data-sources/outputs) in the [Terraform Cloud/Enterprise Provider](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs) to access remote state outputs in Terraform Cloud or Terraform Enterprise. The tfe_outputs data source is more secure because it does not require write access to workspace state to fetch outputs.
```

Author: It seems that I am lacking write access to the repo so here is the PR [#31514](https://github.com/hashicorp/terraform/pull/31514) to update the "update-important-section" branch. Thanks!
